### PR TITLE
Make hotrestart_handoff test also test QUIC

### PR DESCRIPTION
**Pull request by ravenblackx**
Commit Message: Make hotrestart_handoff test also test QUIC
Additional Description: This is still not an adequate test, as the *problem* situation of hot restart handoff is, for the most part, not existing connections continuing to send their outgoing packets (which this test now covers), but rather existing connections that still need to receive more packets from downstream. This addition is still a step in the right direction, but it's not yet enough to provoke failure despite the still incomplete QUIC hot restart support! This change also fixes a pre-existing issue with the test, that it was still doing asynchronous operations sequentially, and adds more traffic to the test to improve the chances that if the behavior is flaky (as it would be with packets being delivered to an arbitrary instance), the test fails.
Risk Level: Test-only.
Testing: Yes it is.
Docs Changes: No.
Release Notes: No.
Platform Specific Features: No.

### CODE REVIEW

File: test/integration/hotrestart_handoff_test.py [hunk 1]
```diff
@@ -58,7 +61,11 @@ def random_loopback_host():
 # the socket path too long.
 SOCKET_PATH = f"@envoy_domain_socket_{os.getpid()}"
 SOCKET_MODE = 0
-ENVOY_BINARY = "./test/integration/hotrestart_main"
+ENVOY_BINARY = ""  # Populated from args
```
Comment (phlax) [inline test/integration/hotrestart_handoff_test.py:64]: are these unused now? if so lets remove its kinda confusing trying to understand what is used where with these here

File: test/integration/hotrestart_handoff_test.py [hunk 2]
```diff
@@ -110,48 +117,102 @@ async def slow_response(self, request):
...
+    def _queue(self) -> asyncio.Queue[str]:
+        return asyncio.Queue()
+
+    @cached_property
+    def _task(self):
+        return asyncio.create_task(self.generator())
+
+    @abc.abstractmethod
+    async def generator(self) -> None:
+        pass
```
Comment (phlax) [inline test/integration/hotrestart_handoff_test.py:132]:
```suggestion
        raise NotImplementedError
```

if its abstract but should always be overridden then rather make that explicit

its kinda annoying that there is not an `interfacemethod` in `std` - `abstracts` (lib in toolshed does)

File: test/integration/hotrestart_handoff_test.py (hunk 1 above)
Comment (phlax) [reply test/integration/hotrestart_handoff_test.py:64]: if these "CONSTANTS" are getting written somewhere and then implicitly used with updated values then even more this needs fixing

File: test/integration/hotrestart_handoff_test.py [hunk 3]
```diff
@@ -110,48 +117,102 @@ async def slow_response(self, request):
...
+        # where connections go - reusing a connection would do the wrong thing.
+        async with ClientSession() as session:
+            async with session.get(self._url) as response:
+                async for line in response.content:
+                    await self._queue.put(line)
+
+
+def _full_http3_request(url: str) -> Awaitable[str]:
+
+    async def req() -> str:
```
Comment (phlax) [inline test/integration/hotrestart_handoff_test.py:183]: closure is almost never a good idea for humans or pooters

Comment (phlax) [reply test/integration/hotrestart_handoff_test.py:183]: (or test engineers for that matter)

**Review by phlax — COMMENTED** [review]
at a high level - code looks amazing - really nice - but tbf would need to test it more actively to get a better idea of running - i guess if it works for intended usage that is probs the acceptance required

python nags/nits inline

(huge apology for slow review)

**Comment by adisuissa** [comment]
@phlax any other comments?

**Comment by phlax** [comment]
no waiting for any other updates

/wait

**Review by phlax — APPROVED** [review]
really nice! lgtm, thanks @ravenblackx

**Pull request merged**